tests/ixia/pfcwd/files/pfcwd_runtime_traffic_helper.py

```diff
@@ -0,0 +1,224 @@
+        data_flow_name (str): data flow name
+        data_flow_dur_sec (int): duration of data flows in second
+        data_pkt_size (int): size of data packets in byte
+        prio_list (list): priorities of data flows
+        prio_dscp_map (dict): Priority vs. DSCP map (key = priority).
+
+    Returns:
+        flows configurations (list): the list should have configurations of
+        len(prio_list) data flows
+    """
+    rx_port_id = port_id
+    tx_port_id = (port_id + 1) % len(testbed_config.devices)
+
+    data_endpoint = DeviceTxRx(
+        tx_device_names=[testbed_config.devices[tx_port_id].name],
+        rx_device_names=[testbed_config.devices[rx_port_id].name],
+    )
+
+    result = list()
+    data_flow_rate_percent = int(100 / len(prio_list))
+
+    """ For each priority """
+    for prio in prio_list:
+        ip_prio = Priority(Dscp(phb=FieldPattern(choice=prio_dscp_map[prio]),
+                                    ecn=FieldPattern(choice=Dscp.ECN_CAPABLE_TRANSPORT_1)))
+        pfc_queue = FieldPattern([prio])
+
+        data_flow = Flow(
+            name='{} Prio {}'.format(data_flow_name, prio),
+            tx_rx=TxRx(data_endpoint),
+            packet=[
+                Header(choice=EthernetHeader(pfc_queue=pfc_queue)),
+                Header(choice=Ipv4Header(priority=ip_prio))
+            ],
+            size=Size(data_pkt_size),
+            rate=Rate('line', data_flow_rate_percent),
+            duration=Duration(FixedSeconds(seconds=data_flow_dur_sec))
+        )
+
+        result.append(data_flow)
+
+    return result
+
+def __run_traffic(api, config, duthost, all_flow_names, pfcwd_start_delay_sec, exp_dur_sec):
+    """
+    Start traffic at time 0 and enable PFC watchdog at pfcwd_start_delay_sec
+
+    Args:
+        api (obj): IXIA session
+        config (obj): experiment config (testbed config + flow config)
+        duthost (Ansible host instance): device under test
+        all_flow_names (list): list of names of all the flows
+        pfcwd_start_delay_sec (int): PFC watchdog start delay in second
+        exp_dur_sec (int): experiment duration in second
+
+    Returns:
+        per-flow statistics (list)
+    """
+
+    api.set_state(State(ConfigState(config=config, state='set')))
+    api.set_state(State(FlowTransmitState(state='start')))
+
+    time.sleep(pfcwd_start_delay_sec)
+    start_pfcwd(duthost)
+    time.sleep(exp_dur_sec - pfcwd_start_delay_sec)
+
+    attempts = 0
+    max_attempts = 20
+
+    while attempts < max_attempts:
+        rows = api.get_flow_results(FlowRequest(flow_names=all_flow_names))
+        """ If all the flows have stopped """
+        transmit_states = [row['transmit'] for row in rows]
+        if len(rows) == len(all_flow_names) and\
+           list(set(transmit_states)) == ['stopped']:
+            time.sleep(IXIA_POLL_DELAY_SEC)
+            break
+        else:
+            time.sleep(1)
+            attempts += 1
+
```

Comment:
we might want to check if we have exhausted all our attempted and could not get the flow results

Reply:
This has been implemented. See 'while attempts < max_attempts'

Reply:
I mean after the while, we need to check if did break out of the loop or executed it till the end otherwise you might not have all the 'rows' as you expected

Reply:
We can always get all the 'rows' as long as IXIA does generate those flows. But if flows do not stop as expected, then __verify_results will report errors.

Reply:
Then we need to assert if we have reached the end of the loop

Reply:
Discussed offline. will add this as a separate PR since there are other test that needs to have the same checks